File: ServiceExampleApp/app/src/main/java/com/example/serviceexampleapp/MainActivity.java
package com.example.serviceexampleapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    private EditText editTextInput;
    private Button start, stop;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editTextInput = findViewById(R.id.editTextInput);
        start = findViewById(R.id.btnStart);
        stop = findViewById(R.id.btnStop);

        start.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String input = editTextInput.getText().toString();
                Intent serviceIntent = new Intent(MainActivity.this, MyService.class);
                serviceIntent.putExtra("inputExtra", input);
                ContextCompat.startForegroundService(MainActivity.this, serviceIntent);
            }
        });

        stop.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent serviceIntent = new Intent(MainActivity.this, MyService.class);
                stopService(serviceIntent);
            }
        });

    }
}

File: SharedPreference Demo in Kotlin/app/src/main/java/com/example/sharedpreferencedemoinkotlin/MainActivity.kt
package com.example.sharedpreferencedemoinkotlin

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val sharedPreferences = getSharedPreferences("UserInfo", Context.MODE_PRIVATE)

        buttonLogin.setOnClickListener {
            val loginId = editTextLoginId.text.toString()
            val password = editTextPassword.text.toString()
            val keepLoggedIn = checkBoxKeepLoggedIn.isChecked

            Toast.makeText(this, "Successfully LoggedIn!!", Toast.LENGTH_LONG)

            val editor = sharedPreferences.edit()
            editor.putString("id", loginId)
            editor.putString("pass", password)
            editor.putBoolean("check", keepLoggedIn)
            editor.apply()
        }

        buttonShowDetails.setOnClickListener {
            val loginId = sharedPreferences.getString("id", "")
            val password = sharedPreferences.getString("pass", "")
            val keepLoggedIn = sharedPreferences.getBoolean("check", false)

            textViewShowDetail.text =
                "Email : $loginId \nPassword : $password\nSave Credentials To login Again : $keepLoggedIn "
        }
    }
}
